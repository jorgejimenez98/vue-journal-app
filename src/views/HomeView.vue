<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
  </div>

  <button @click="goToDaybook" type="button" class="btn btn-primary">Day Book</button>
  <button type="button" class="btn btn-secondary">Primary</button>
  <button type="button" class="btn btn-warning">Primary</button>
</template>

<script>
export default {
  methods: {
    goToDaybook() {
      this.$router.push({ name: "no-entry" });
    },
  },
};
</script>
