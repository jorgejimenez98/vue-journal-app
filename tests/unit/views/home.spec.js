import { shallowMount } from "@vue/test-utils";
import HomeView from "@/views/HomeView.vue";

describe("HomeView.vue", () => {
  it("HomeView must match on snapshot", () => {
    const wrapper = shallowMount(HomeView);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("hacsr click en un boton debe de redireccionar al no-entry", () => {
    const mockRouter = {
      push: jest.fn(),
    };

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalled();
    // Evaluar el param del router
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
