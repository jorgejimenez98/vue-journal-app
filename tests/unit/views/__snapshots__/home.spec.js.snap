// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue HomeView must match on snapshot 1`] = `<div class="home"><img alt="Vue logo" src=""></div><button type="button" class="btn btn-primary">Day Book</button><button type="button" class="btn btn-secondary">Primary</button><button type="button" class="btn btn-warning">Primary</button>`;
